### PR TITLE
fix: Download lambda

### PR DESCRIPTION
**Pull request by npalm**
Changes applied in PR #1478 are not working properly.

## Description
Function `sync` is not triggered via the Lambda handler. Cause uploadS3 function was not awaited. Also the s3upload function was not tested. 

## Solution
- add await
- fix tests
- adjust coverage criteria

### CODE REVIEW

**Review by ScottGuymer — DISMISSED** [review]
LGTM :shipit:

**Comment by npalm** [comment]
@ScottGuymer @PatrickMennen please can you check the change

File: modules/runner-binaries-syncer/lambdas/runner-binaries-syncer/src/syncer/syncer.test.ts [hunk 1]
```diff
@@ -13,9 +17,26 @@ jest.mock('@octokit/rest', () => ({
...
+jest.mock('axios');
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+mockedAxios.request.mockResolvedValue({
+  data: mockStream,
+});
+
 const mockS3 = {
   getObjectTagging: jest.fn(),
-  upload: jest.fn(),
+  // upload: jest.fn(() => {
```
Comment (ScottGuymer) [inline modules/runner-binaries-syncer/lambdas/runner-binaries-syncer/src/syncer/syncer.test.ts:34]: Remove the commented out code.

File: modules/runner-binaries-syncer/lambdas/runner-binaries-syncer/src/syncer/syncer.test.ts [hunk 2]
```diff
@@ -190,8 +213,8 @@ describe('Synchronize action distribution.', () => {
       Key: bucketObjectKey,
     });
     expect(mockS3.upload).toBeCalledTimes(1);
-    const s3JsonBody = mockS3.upload.mock.calls[0][0];
-    expect(s3JsonBody['Tagging']).toEqual('name=actions-runner-linux-x64-2.273.0.tar.gz');
+    //const s3JsonBody = mockS3.upload.mock.calls[0][0];
```
Comment (ScottGuymer) [inline modules/runner-binaries-syncer/lambdas/runner-binaries-syncer/src/syncer/syncer.test.ts:216]: Remove commented out code

**Review by ScottGuymer — APPROVED** [review]

**Pull request merged**